frontend/src/components/shared/CsgButton.vue

```diff
@@ -31,6 +37,70 @@
     loading: {
       type: Boolean,
       default: false
+    },
+    disabled: {
+      type: Boolean,
+      default: false
+    }
+  });
+
+  // Map the reserved keyword 'class' to a regular variable 'className'
+  const className = computed(() => props.class);
+
+  // Check if the class contains disabled class
+  const isDisabledClass = computed(() => {
+    return props.class.includes('disabled');
+  });
+
+  // Calculate icon size based on button size
+  const getIconSizeClass = computed(() => {
+    if (props.class.includes('btn-2xl')) {
+      return 'w-6 h-6'; // 24px
+    } else if (props.class.includes('btn-xl')) {
+      return 'w-[18px] h-[18px]'; // 18px
+    } else if (props.class.includes('btn-lg')) {
+      return 'w-4 h-4'; // 16px
+    } else if (props.class.includes('btn-md')) {
+      return 'w-3.5 h-3.5'; // 14px
+    } else if (props.class.includes('btn-sm')) {
+      return 'w-3 h-3'; // 12px
+    } else {
+      return 'w-4 h-4'; // Default size
     }
   });
 </script>
+
+<style scoped>
+  /* Use stronger selectors to ensure SVG colors match button text */
+  button :deep(svg),
+  button :deep(svg path),
+  button :deep(svg rect),
+  button :deep(svg circle),
+  button :deep(svg *) {
+    fill: currentColor !important;
+    color: inherit !important;
+    stroke: currentColor !important;
+  }
+  
+  /* Special handling for disabled state */
+  button[data-disabled="true"] :deep(svg),
+  button[data-disabled="true"] :deep(svg path),
+  button[data-disabled="true"] :deep(svg rect),
+  button[data-disabled="true"] :deep(svg circle),
+  button[data-disabled="true"] :deep(svg *) {
+    fill: currentColor !important;
+    color: inherit !important;
+    stroke: currentColor !important;
+  }
+
+  /* Add to global CSS file */
+  .btn svg,
+  .btn svg path,
+  .btn svg rect,
+  .btn svg circle,
+  .btn svg * {
+    fill: currentColor !important;
+    color: inherit !important;
+    stroke: currentColor !important;
+  }
+</style>
```

Comment:
as we're using embeded way in SvgIcon, like below, the settings for SVG element will not take effect
```
  <img
    :width="width"
    :height="height"
    :src="`/images/${path}/${name}.svg`"
    onerror="this.style.display='none'"
  />
```